### PR TITLE
Cooking Fixes Pass 2

### PR DESCRIPTION
**Pull request by NanakoAC**
This PR is a package of fixes for a few cooking issues, and also (largely unrelated but its been annoying me for weeks) more fixing on the Flash issue.

Fixes the cooking sound not being set on cookers

Very thoroughly re-examines all interactions with cooking machines and their verbs, adds safety checks, alters contexts as intended. This gets rid of all the problems with mouse/ghost/AI fuckery.
Note that AIs can toggle the oven door and the power on cooking machines, this is by design.
Animals can remove food from exposed cooking containers, this is also by design. 

Sets cooking containers to NOREACT, to prevent some issues with recipes being unmakeable because ingredients would become dough.

Tweaks oven heat loss a little more

Adjusts the mix flag on several potentially problematic recipes to prevent certain undesireable things ending up in finished products (like flour and eggs causing a lump of dough to be ejected from a finished cake)

Alters quite a few recipes which used a whole egg as an ingredient, to use egg yolk instead. Partly becase it makes more sense in context, and partly because cooking containers don't support adding whole eggs. There are now no fryer/oven recipes that need an intact egg.

Scales up plain bread to be a little closer to the other specialised breads in terms of reagent quantity.

Removes a redundant spellburger recipe because item checks use istype

And the flash issue:
For those who missed it, several months ago we had an issue with an object named "flash" appearing in the rightclick menu anywhere, for anyone with admin/dev powers. It was constantly annoying. After enduring the mysterious thing for months i found the cause, it's a hud element. Setting mouse opacity fixed it.

Anyways my fix wasn't comprehensive enough. Several other huds for non human mobs duplicate this thing. So ive set mouse opacity on them too

### CODE REVIEW

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -3002,14 +3002,15 @@
...
-	slices_num = 5
+	slices_num = 8
 	filling_color = "#FFE396"
 	center_of_mass = list("x"=16, "y"=9)
 
-/obj/item/weapon/reagent_containers/food/snacks/sliceable/bread/New()
-	..()
-	reagents.add_reagent("nutriment", 6)
-	bitesize = 2
+	New()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Use full paths

File: code/game/machinery/kitchen/cooking_machines/oven.dm [hunk 2]
```diff
@@ -50,24 +50,38 @@
...
 /obj/machinery/appliance/cooker/oven/AltClick(var/mob/user)
+	.=1
 	if(user.stat || user.restrained())	return
 	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)//No spamming the door, it makes a sound
 	toggle_door()
 
+
 /obj/machinery/appliance/cooker/oven/verb/toggle_door()
-	set src in view()
+	set src in world //So that AI can operate it remotely
```
Comment (Arrow768) [inline code/game/machinery/kitchen/cooking_machines/oven.dm]: I dont like this for the following reasons:
AIs can use alt-click
AIs can use rightclick -> Open / Close oven door
What if we decide to add more than one oven at a later point ?

Therefore I think AIs can live without having the Open/Close oven door verb in their object tab

File: code/game/machinery/kitchen/cooking_machines/_appliance.dm [hunk 3]
```diff
@@ -110,14 +110,18 @@
 		icon_state = off_icon
 
 /obj/machinery/appliance/verb/toggle_power()
-	set src in view()
 	set name = "Toggle Power"
 	set category = "Object"
+	set src in world //So that AI can operate it remotely
```
Comment (Arrow768) [inline code/game/machinery/kitchen/cooking_machines/_appliance.dm]: Same here. Add it to the right click menu but remove it from the object tab.
"Toggle Power" in the object tab of an AI can mean anything.

**Comment by NanakoAC** [comment]
I don't understand your objections. 

You're still only going to see the verb if you rightclick on it

**Comment by Lohikar** [comment]
The verb will be listed in the Objects stat panel for everyone is what I believe they are saying.

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (NanakoAC) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: This is how the cooking files are done, for good reason. ITs an insanely huge mess of many objects, and is already painful to navigate. Almost nothing in there uses full paths, and adding them would make the problem significantly worse

File: code/game/machinery/kitchen/cooking_machines/oven.dm (hunk 2 above)
Comment (NanakoAC) [inline code/game/machinery/kitchen/cooking_machines/oven.dm]: It does not appear in the rightclick menu without doing this. This is what i had to do to make it appear there

**Comment by skull132** [comment]
What Lohikar said.

File: code/game/machinery/kitchen/cooking_machines/container.dm [hunk 4]
```diff
@@ -48,13 +48,23 @@
 	set category = "Object"
 	set desc = "Removes items from the container. does not remove reagents."
 
+	if (!isliving(usr))
+		usr << "Ghosts can't mess with cooking containers"
```
Comment (Lohikar) [inline code/game/machinery/kitchen/cooking_machines/container.dm]: Do we really need to give a feedback message to ghosts? Most things don't have one.

File: code/game/machinery/kitchen/cooking_machines/oven.dm (hunk 2 above)
Comment (Lohikar) [inline code/game/machinery/kitchen/cooking_machines/oven.dm]: ...why can the AI open the oven door anyways? Isn't it something that requires physical interaction with the appliance?

**Comment by NanakoAC** [comment]
Well i've actually gone and tested it, seems you guys' assumption was incorrect

despite the src in world setting, the toggle power verb only appears - for both the command line and the object panel, when you're somewhere near the kitchen. What exactly near means is a little difficult to determine, but it seems to be roughly up to 20 tiles away.

So it shows up in some parts of the hallway, bar, edge of medical, holodeck, etc but not in most of the station.

I'd say this isn't a problem given that it has safety checks anyways. And if you type "toggle " into the command line (you have to hit spacebar before it starts making suggestions) you'll get a very long list of toggle X commands from various parts of the stat panel anyways. One extra thing in there under certain conditions is hardly an issue

**Comment by Lohikar** [comment]
I don't see why it needs to be a global verb in the first place.
Why can't you just define ctrl+click behavior for the AI?

**Comment by NanakoAC** [comment]
consistency, useability, and AIs not being able to use verbs is dumb.
This method works with minimal clutter

File: code/game/machinery/kitchen/cooking_machines/oven.dm (hunk 2 above)
Comment (NanakoAC) [inline code/game/machinery/kitchen/cooking_machines/oven.dm]: electronic switch

**Comment by Arrow768** [comment]
My main point has been addressed, but I think a GUI would be nice for the devices.
Throw in the current temperature.
Add a button to start and stop it.
And a few buttons to remove the items from it.
And maybe add the time the items have spent in there as a QoL feature

**Comment by skull132** [comment]
> But in any case, I can set the category to null which will cause it to never appear in the upper right panel. It'd only show up in command line, which as mentioned is kinda cluttered anyways, and only in conditions described above. I'd argue this would have only a neglible impact on a tiny minority of users - mostly us tbh. Command line verbs aren't commonly used by players

I hope you could hear me groan as you wrote that. Because I do hate the argument. Also, the command-line is a great way to play fast. Cluttering it with generic verbs is bad, and the reason of, "It's already bad, so might as well make it more bad," is also, bad. And on the flipside, having the category set as object instead of null is great for indicating that there's new interaction available for people that do rely on the tabs. Win-win.

Why not just add an entry-way from one of the click combinations for an AI to use?

**Comment by Lohikar** [comment]
It is possible to add a verb only to a specific type of mob, but it would require that the AI get its own verb, and the method is used nowhere else.

I still feel that cooking appliances should use a UI (preferably NanoUI or the browser datum) instead of verbs.

**Comment by NanakoAC** [comment]
ok power toggling for AI is now control+click, verbs returned to in view

File: code/game/machinery/kitchen/cooking_machines/_appliance.dm [hunk 5]
```diff
@@ -112,7 +112,7 @@
 /obj/machinery/appliance/verb/toggle_power()
 	set name = "Toggle Power"
 	set category = null
```
Comment (Lohikar) [inline code/game/machinery/kitchen/cooking_machines/_appliance.dm]: Why not make robots use the AI verb and make this `src in view(1)`? That'll reduce verb clutter further.
Also, probably should remove `set category = null` here.

File: code/game/machinery/kitchen/cooking_machines/oven.dm [hunk 6]
```diff
@@ -57,7 +57,7 @@
 
 
 /obj/machinery/appliance/cooker/oven/verb/toggle_door()
-	set src in world //So that AI can operate it remotely
+	set src in view()
```
Comment (Lohikar) [inline code/game/machinery/kitchen/cooking_machines/oven.dm]: Same here.

**Pull request merged**